packages/datatrak-web-server/src/utils/processSurveyResponse.ts

```diff
@@ -54,13 +57,23 @@ export const processSurveyResponse = async (
     entities_upserted: [],
     options_created: [],
     answers: [],
-  } as MeditrakSurveyResponseRequest;
+  } as MeditrakSurveyResponseRequest & {
+    entities_upserted: Entity[];
+  };
   // Process answers and save the response in the database
   const answersToSubmit = [] as Record<string, unknown>[];
 
   for (const question of questions) {
     const { questionId, type } = question;
-    const answer = answers[questionId];
+    let answer = answers[questionId] as AnswerT | Entity;
+    const config = question?.config as ConfigT;
+
+    // If the question is an entity question and an entity should be created by this question, build the entity object. We need to do this before we get to the check for the answer being empty, because most of the time these questions are hidden and therefore the answer will always be empty
+    if (['PrimaryEntity', 'Entity'].includes(type) && isUpsertEntityQuestion(config)) {
```

Comment:
```suggestion
    if ([QuestionType.PrimaryEntity, QuestionType.Entity] && isUpsertEntityQuestion(config)) {
```
Might as well use the enum here for consistency